File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        # Using array 
        # ans = []
        # i, j = 0, 0

        # while i < len(word1) and j < len(word2):
        #     ans.append(word1[i])
        #     ans.append(word2[j])
        #     i +=1
        #     j +=1
        
        # ans.append(word1[i:])
        # ans.append(word2[j:])
        # return ''.join(ans)


        # using string operations
        # i=0
        # j=0
        # word3 = ''
        # while((i<len(word1) and (j<len(word2)))):
        #     word3 = word3 + word1[i] + word2[j]
        #     i += 1
        #     j += 1
        # while(i<len(word1)):
        #     word3 = word3 + word1[i]
        #     i += 1
        # while(j<len(word2)):
        #     word3 = word3 + word2[j]
        #     j += 1
        # return word3

        # Using min() and Slicing
        n1 = len(word1)
        n2 = len(word2)
        result = ""
        for i in range(0,min(n1,n2)):
            result = result + word1[i] + word2[i]

        if(n1 < n2):
            result = result + word2[n1:]
        else:
            result = result + word1[n2:]
        
        return result

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {

        // StringBuilder result = new StringBuilder();
        
        // int i = 0;
        // while (i < word1.length() || i < word2.length()){
        //     if(i < word1.length()){
        //         result.append(word1.charAt(i));
        //     }
        //     if(i < word2.length()){
        //         result.append(word2.charAt(i));
        //     }
        //     i++;
        // }

        // for (int i =0; i < word1.length() || i < word2.length(); i++){
        //     if(i < word1.length()){
        //         result.append(word1.charAt(i));
        //     }
        //     if(i < word2.length()){
        //         result.append(word2.charAt(i));
        //     }
        // }
        // return result.toString();


        String result = "";
        int maxLength = Math.max(word1.length(), word2.length());
        for (int i = 0; i < maxLength; i++) {
            if (i < word1.length()) {
                result += word1.charAt(i);
            }
            if (i < word2.length()) {
                result += word2.charAt(i);
            }
        }
        return result;

    }
}

